### PR TITLE
snap-mgmt: extend spread tests, stop, disable and cleanup snap services

### PR DESCRIPTION
**Pull request by bboozzoo**
Make sure we clean up stuff while doing purge.

### CODE REVIEW

File: cmd/snap-mgmt/snap-mgmt.sh.in [hunk 1]
```diff
@@ -36,6 +36,16 @@ purge() {
     # debian-9, fedora-27, ubuntu-16.04...
     distribution=$(. /etc/os-release; echo "${ID}-${VERSION_ID}")
 
+    snap_services=$(systemctl list-unit-files --no-legend --type service | \
+                        awk '/^snap\..*\.service/ { print $1 }' || true)
+    for service in $snap_services; do
+        # stop and disable services that are coming from snaps
+        echo "Stopping snap service $service"
+        systemctl_stop $service || true
```
Comment (zyga) [inline cmd/snap-mgmt/snap-mgmt.sh.in]: Nitpick: `"$service"`

File: cmd/snap-mgmt/snap-mgmt.sh.in [hunk 2]
```diff
@@ -36,6 +36,16 @@ purge() {
...
+        systemctl disable --no-reload $service || true
```
Comment (zyga) [inline cmd/snap-mgmt/snap-mgmt.sh.in]: `"$service"`

File: cmd/snap-mgmt/snap-mgmt.sh.in [hunk 3]
```diff
@@ -36,6 +36,16 @@ purge() {
...
+        rm -f /etc/systemd/system/$service
```
Comment (zyga) [inline cmd/snap-mgmt/snap-mgmt.sh.in]: `/etc/systemd/system/"$service"`

**Comment by bboozzoo** [comment]
I'd say best leave the reload to the caller. `snap-mgmt` might be called from package manager (well it is actually) so calling reload may have some undesirable side effects. Depending on the distro there might also be policies in place that allow calling systemd reload only only in the final cleanup phase, so ideally after calling `snap-mgmt --purge`.

File: cmd/snap-mgmt/snap-mgmt.sh.in [hunk 4]
```diff
@@ -48,8 +48,10 @@ purge() {
     fi
 
     units=$(systemctl list-unit-files --no-legend --full | grep -vF snap.mount.service || true)
+    # *.snap mount points
     mounts=$(echo "$units" | grep "^${SNAP_UNIT_PREFIX}[-.].*\.mount" | cut -f1 -d ' ')
-    services=$(echo "$units" | grep "^${SNAP_UNIT_PREFIX}[-.].*\.service" | cut -f1 -d ' ')
+    # services from snaps
+    services=$(echo "$units" | grep "^snap\..*\.service" | cut -f1 -d ' ')
```
Comment (mvo5) [inline cmd/snap-mgmt/snap-mgmt.sh.in]: Nice catch!

**Pull request merged**